### PR TITLE
refactor: Not use 'Ticket' as a generic term in the app

### PR DESCRIPTION
**Pull request by gorandalum**
'Ticket' was used both for specifying bought fare contracts and available fare products. With this change we instead use the terms which is also used in Firestore and Entur.

Resolves https://github.com/AtB-AS/kundevendt/issues/2663

### CODE REVIEW

File: src/translations/screens/Ticket.ts [hunk 1]
```diff
@@ -143,4 +143,4 @@ const TicketTexts = {
   },
 };
 
-export default TicketTexts;
+export default TicketingTexts;
```
Comment (Nisha1306) [inline src/translations/screens/Ticket.ts:146]: Also change the file name?

File: src/translations/screens/Tickets.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@ const TicketsTexts = {
   header: {
     title: _('Billetter', 'Tickets'),
   },
-  buyTicketsTab: {
+  purchaseTab: {
```
Comment (Nisha1306) [inline src/translations/screens/Tickets.ts:10]: Should we also change main object TicketsTexts to something else here? This and the other TicketTexts (now TicketingText) has always confused me 😅

File: src/screens/Ticketing/FareProducts/types.ts [hunk 3]
```diff
@@ -9,24 +9,25 @@ import {
...
 
 export type TicketsStackRootNavigationProps =
   TabNavigatorNavigationProps<'Ticketing'>;
 
-export type TicketsNavigationProps<T extends keyof TicketTabsNavigatorParams> =
-  CompositeNavigationProp<
-    NavigationProp<TicketTabsNavigatorParams, T>,
-    TicketsStackRootNavigationProps
-  >;
+export type TicketsNavigationProps<
```
Comment (Nisha1306) [inline src/screens/Ticketing/FareProducts/types.ts:22]: Should this be TicketingNavigationProps?

File: src/screens/Ticketing/FareProducts/types.ts [hunk 4]
```diff
@@ -9,24 +9,25 @@ import {
...
 
-export type TicketTabsNavigatorParams = {
-  BuyTickets: undefined;
-  ActiveTickets: undefined;
+export type TicketingTabsNavigatorParams = {
+  PurchaseTab: undefined;
+  ActiveFareProductsAndReservationsTab: undefined;
 };
 
 export type TicketsStackRootProps = TabNavigatorScreenProps<'Ticketing'>;
```
Comment (Nisha1306) [inline src/screens/Ticketing/FareProducts/types.ts:17]: Should this be TicketingStackRootProps?

File: src/screens/Ticketing/FareProducts/types.ts [hunk 5]
```diff
@@ -9,24 +9,25 @@ import {
...
-export type TicketsNavigationProps<T extends keyof TicketTabsNavigatorParams> =
-  CompositeNavigationProp<
-    NavigationProp<TicketTabsNavigatorParams, T>,
-    TicketsStackRootNavigationProps
-  >;
+export type TicketsNavigationProps<
+  T extends keyof TicketingTabsNavigatorParams,
+> = CompositeNavigationProp<
+  NavigationProp<TicketingTabsNavigatorParams, T>,
+  TicketsStackRootNavigationProps
```
Comment (Nisha1306) [inline src/screens/Ticketing/FareProducts/types.ts:26]: TicketingStackRootNavigationProps?

File: src/screens/Ticketing/FareProducts/RecentFareProducts/RecentFareContracts.tsx [hunk 6]
```diff
@@ -94,18 +94,18 @@ export const RecentTickets = () => {
...
-                ticket.userProfilesWithCount
+                rfc.preassignedFareProduct.id +
+                rfc.userProfilesWithCount
                   .map((traveller) => {
                     return traveller.count + traveller.userTypeString;
                   })
                   .join();
               return (
-                <RecentTicketComponent
+                <RecentFareContractComponent
```
Comment (Nisha1306) [inline src/screens/Ticketing/FareProducts/RecentFareProducts/RecentFareContracts.tsx:106]: Think we dont really need to append **component** at the end.

File: src/ticketing/TicketingContext.tsx [hunk 7]
```diff
@@ -35,31 +35,31 @@ type TicketReducerAction =
...
 
-const ticketReducer: TicketReducer = (
+const ticketingReducer: TicketingReducer = (
   prevState,
   action,
-): TicketReducerState => {
+): TicketingReducerState => {
   switch (action.type) {
-    case 'SET_IS_REFRESHING_FARE_CONTRACT_TICKETS': {
+    case 'SET_IS_REFRESHING_FARE_CONTRACTS': {
```
Comment (Nisha1306) [inline src/ticketing/TicketingContext.tsx:48]: I don't see this being triggered from anywhere, looks redundant 🤔

File: src/ticketing/utils.ts [hunk 8]
```diff
@@ -1,40 +1,40 @@
...
   );
 }
 
-export function isSingleTicket(
+export function isSingleTravelRight(
   travelRight: TravelRight | undefined,
-): travelRight is PreactivatedSingleTicket {
+): travelRight is PreActivatedSingleTravelRight {
   return travelRight?.type === 'PreActivatedSingleTicket';
 }
```
[the remark this answers is not in the document]
Comment (gorandalum) [reply src/ticketing/utils.ts:35]: You are right. However, I think I don't do anything about this now as I want to get rid of all this ´PreactivatedSingleTicket`, `PreactivatedCarnetTicket` stuff etc. pretty soon. 

I don't actually now what `Preactivated` is ment to be used for, and the type of the fare contract should rather be retrieved through the `type` on the `PreassignedFareProduct` by following the `preassignedFareProductRef` on the `FareContract`.

File: src/screens/Ticketing/FareProducts/RecentFareProducts/RecentFareContracts.tsx (hunk 6 above)
Comment (gorandalum) [reply src/screens/Ticketing/FareProducts/RecentFareProducts/RecentFareContracts.tsx:106]: Hmm, this kinda triggered an interesting question. We have a type named `RecentFareContract`, and if renaming this we also have a component named `RecentFareContract`. They are in naming and casing exactly the same, and will be hard to distinguish in the code. I think we need an established convention here, so until we have it, I'll just keep the `RecentFareContractComponent` name.

Comment (Nisha1306) [reply src/screens/Ticketing/FareProducts/RecentFareProducts/RecentFareContracts.tsx:106]: Ohh okie,  I would at least like to rename the type in such cases and keep the Component as it is. But lets leave as is for now 👍

**Comment by Nisha1306** [comment]
@gorandalum  I still see one screen name TicketPurchase used in a lot of places. Maybe rename that as well?

**Comment by gorandalum** [comment]
> @gorandalum I still see one screen name TicketPurchase used in a lot of places. Maybe rename that as well?

Done now 👍 Just using `Purchase` for now, so we'll need to rename when we starts selling other stuff in the app 😅

**Comment by gorandalum** [comment]
I'll wait to merge this until a release 1.27 branch is created.

**Review by Nisha1306 — APPROVED** [review]

**Pull request merged**